server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGateUpdater.java

```diff
@@ -73,15 +68,9 @@ private void validateQualityGate(DbSession dbSession, OrganizationDto organizati
     checkQualityGateDoesNotAlreadyExist(dbSession, organizationDto, name);
   }
 
-  public void setDefault(DbSession dbSession, @Nullable QualityGateDto qualityGateDto) {
-    if (qualityGateDto == null) {
-      dbClient.propertiesDao().deleteGlobalProperty(SONAR_QUALITYGATE_PROPERTY, dbSession);
-    } else {
-      long qualityGateId = qualityGateDto.getId();
-      checkFound(dbClient.qualityGateDao().selectById(dbSession, qualityGateId), "There is no quality gate with id=" + qualityGateId);
-      dbClient.propertiesDao().saveProperty(dbSession,
-        new PropertyDto().setKey(SONAR_QUALITYGATE_PROPERTY).setValue(qualityGateDto.getId().toString()));
-    }
+  public void setDefault(DbSession dbSession, OrganizationDto organizationDto, QualityGateDto qualityGateDto) {
+    organizationDto.setDefaultQualityGateUuid(qualityGateDto.getUuid());
```

Comment:
It should be best to have only one way to set the default quality gate on an organization : either we use OrganizationDao#setDefaultQualityGate, or OrganizationDao#update, but not both